src/main/java/org/spongepowered/common/util/gen/ArrayMutableBlockBuffer.java

```diff
@@ -39,25 +40,28 @@
 import org.spongepowered.common.world.extent.MutableBlockViewTransform;
 import org.spongepowered.common.world.extent.UnmodifiableBlockVolumeWrapper;
 import org.spongepowered.common.world.extent.worker.SpongeMutableBlockVolumeWorker;
+import org.spongepowered.common.world.schematic.BimapPalette;
+import org.spongepowered.common.world.schematic.GlobalPalette;
 
 public class ArrayMutableBlockBuffer extends AbstractBlockBuffer implements MutableBlockVolume {
 
+    private static final int LOCAL_PALETTE_THRESHOLD = 256;
+
     @SuppressWarnings("ConstantConditions")
     private static final BlockState AIR = BlockTypes.AIR.getDefaultState();
 
     private final BlockPalette palette;
     private final BackingData data;
 
-    public ArrayMutableBlockBuffer(BlockPalette palette, Vector3i start, Vector3i size, BackingDataType type) {
-        super(start, size);
-        this.palette = palette;
-        this.data = type.create(size);
+    public ArrayMutableBlockBuffer(Vector3i start, Vector3i size) {
+        this(size.getX() * size.getY() * size.getZ() > LOCAL_PALETTE_THRESHOLD ?
+                new BimapPalette() : GlobalPalette.instance, start, size);
```

Comment:
I know its not the cause of bug 1. But you make a good point, I did not think about the fact that the buffer defaults to being filled with air blocks.
(In the images of my test plugin, setBlock() was called on all block postions, so there where no *undefined* blocks)